src/math/p5.Vector.js

```diff
@@ -326,6 +326,141 @@ p5.Vector.prototype.add = function add(x, y, z) {
   return this;
 };
 
+/**
+ * Gives remainder of a vector when it is divided by another vector.
+ * See examples for more context.
+ *
+ * @method rem
+ * @param {Number} x the x component of divisor vector
+ * @param {Number} y the y component of divisor vector
+ * @param {Number} z the z component of divisor vector
+ * @chainable
+ * @example
+ * <div class='norender'>
+ * <code>
+ * let v = createVector(3, 4, 5);
+ * v.rem(2, 3, 4);
+ * // v's components are set to [1, 1, 1]
+ * </code>
+ * </div>
+ * <div class="norender">
+ * <code>
+ * // Static method
+ * let v1 = createVector(3, 4, 5);
+ * let v2 = createVector(2, 3, 4);
+ *
+ * let v3 = p5.Vector.rem(v1, v2);
+ * // v3 has components [1, 1, 1]
+ * print(v3);
+ * </code>
+ * </div>
+ */
+/**
+ * @method rem
+ * @param {p5.Vector | Number[]}  value  divisor vector
+ * @chainable
+ */
+p5.Vector.prototype.rem = function rem(x, y, z) {
+  const calculateRemainder3D = (xComponent, yComponent, zComponent) => {
+    if (xComponent === 0 && yComponent === 0 && zComponent === 0) {
+      return this;
+    }
+    if (xComponent === 0 && yComponent !== 0 && zComponent !== 0) {
+      this.y = this.y % yComponent;
+      this.z = this.z % zComponent;
+      return this;
+    }
+    if (xComponent !== 0 && yComponent === 0 && zComponent !== 0) {
+      this.x = this.x % xComponent;
+      this.z = this.z % zComponent;
+      return this;
+    }
+    if (xComponent !== 0 && yComponent !== 0 && zComponent === 0) {
+      this.x = this.x % xComponent;
+      this.y = this.y % yComponent;
+      return this;
+    }
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    this.z = this.z % zComponent;
+    return this;
+  };
+
+  const calculateRemainder2D = (xComponent, yComponent) => {
+    if (xComponent === 0 && yComponent === 0) {
+      return this;
+    }
+    if (xComponent === 0 && yComponent !== 0) {
+      this.y = this.y % x[1];
+      return this;
+    }
+    if (xComponent !== 0 && yComponent === 0) {
+      this.x = this.x % x[0];
+      return this;
+    }
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    return this;
+  };
+
+  if (x instanceof p5.Vector) {
+    if (Number.isFinite(x.x) && Number.isFinite(x.y) && Number.isFinite(x.z)) {
+      var xComponent = parseFloat(x.x);
+      var yComponent = parseFloat(x.y);
+      var zComponent = parseFloat(x.z);
+      calculateRemainder3D(xComponent, yComponent, zComponent);
+    }
+  }
+
+  if (x instanceof Array) {
+    if (x.every(element => Number.isFinite(element))) {
+      if (x.length === 2) {
+        calculateRemainder2D(x[0], x[1]);
+      }
+      if (x.length === 3) {
+        calculateRemainder3D(x[0], x[1], x[2]);
+      }
+    }
+  }
+
+  if (arguments.length === 0) {
+    if (arguments[0] === undefined) {
```

Comment:
can you explain this check?

Reply:
Looked at the code again and removed this block and tests are running !